flight/Modules/Actuator/actuator.c

```diff
@@ -191,76 +214,34 @@ static void actuatorTask(void* parameters)
 	{
 		PIOS_WDG_UpdateFlag(PIOS_WDG_ACTUATOR);
 
-		// Wait until the ActuatorDesired object is updated
-		bool rc = PIOS_Queue_Receive(queue, &ev, FAILSAFE_TIMEOUT_MS);
+		bool rc=false;
 
-		/* Process settings updated events even in timeout case so we always act on the latest settings */
-		if (actuator_settings_updated) {
-			actuator_settings_updated = false;
-			ActuatorSettingsGet (&actuatorSettings);
-			actuator_update_rate_if_changed (&actuatorSettings, false);
+		// If it's our first iteration, fall down to getting the
+		// config and enabling failsafe.
+		if (!first) {
```

Comment:
<facepalm>

Totally my fault. I missed that there was another function called ```SettingsUpdatedCb()```, which is where the confusion came from. I think you got what I meant in the end.

The advantage of the call before is we can eliminate the ```first``` variable and its conditional. I like to reduce complexity inside the while loop, not only for efficiency reasons but because it makes it easier to understand what the loop is intended to do. I admit to getting a little carried away by being offended at the notion of running something once, and then checking forever afterwards. It feels icky, but it truly is not a big impact on processor.